invenio_communities/communities/service.py

```diff
@@ -10,10 +10,62 @@
 
 """Invenio Communities Service API."""
 
+from invenio_records_resources.services.base import LinksTemplate
+from invenio_db import db
 from invenio_records_resources.services.records import RecordService
+from marshmallow.exceptions import ValidationError
 
 
 class CommunityService(RecordService):
     """community Service."""
 
-    pass
+    def search_user_communities(
+            self, identity, params=None, es_preference=None, **kwargs):
+        """Search for records matching the querystring."""
+        self.require_permission(identity, 'search')
```

Comment:
Shouldn't this be `search_user_communities` (which should then be set to `AuthenticatedUser()`)?

Then the `permission_action` for the resource should be just `read` (same way as it's done in drafts: https://github.com/inveniosoftware/invenio-drafts-resources/blob/master/invenio_drafts_resources/services/records/service.py#L49-L65)